k3k-kubelet/kubelet.go

```diff
@@ -273,35 +282,40 @@ func loadTLSConfig(ctx context.Context, hostClient ctrlruntimeclient.Client, clu
 	if err := hostClient.Get(ctx, types.NamespacedName{Name: clusterName, Namespace: clusterNamespace}, &cluster); err != nil {
 		return nil, err
 	}
-	endpoint := fmt.Sprintf("%s.%s", server.ServiceName(cluster.Name), cluster.Namespace)
+
+	endpoint := server.ServiceName(cluster.Name) + "." + cluster.Namespace
 	if err := retry.OnError(controller.Backoff, func(err error) bool {
 		return err != nil
 	}, func() error {
 		var err error
 		b, err = bootstrap.DecodedBootstrap(token, endpoint)
 		return err
 	}); err != nil {
-		return nil, fmt.Errorf("unable to decode bootstrap: %w", err)
+		return nil, err
 	}
```

Comment:
Since we are using this func a lot maybe it's more clear if we extract this in a common/util/error package:

```go
func IsNotNil(err error) bool {
    return err != nil
}
```

Then if we want to chain the err in the same line I like more to have short definition, so something like:

```go
decode := func() error {
	var err error
	b, err = bootstrap.DecodedBootstrap(token, endpoint)
	return err
}
if err := retry.OnError(controller.Backoff, errors.IsNotNil, decode); err != nil {
	return nil, err
}
```

 Just a personal nit. And not sure about the scope of `b` in the closure!

Reply:
I'd probably be beneficial to move this out to another function but more so in another PR.